streams/src/main/java/org/apache/kafka/streams/kstream/internals/SerializedInternal.java

```diff
@@ -19,6 +19,7 @@
 import org.apache.kafka.common.serialization.Serde;
 import org.apache.kafka.streams.kstream.Serialized;
 
+@Deprecated
```

Comment:
No need to deprecate an internal class IMHO. Maybe add `@SuppressWarning("deprecation")` instead?

Reply:
This is the same thinking as https://github.com/apache/kafka/pull/5709#discussion_r221462040 .